group_project_v2/mixins.py

```diff
@@ -125,7 +126,180 @@ def real_user_id(self, anonymous_student_id):
+                 view user will need to belong to a team. If user is a member
+                 of this group he will be able to access group work for every
+                 group work team.
+        :rtype: Iterable[str]
+        """
+        return self._get_setting(self._TA_ROLE, self._DEFAULT_TA_ROLE)
+
+    def can_access_dashboard(self, user_id):
+        """
+        :param user_id:
+        :return: True if user can access dashboard.
+        """
+        user_groups = self._user_groups(user_id)
+        return bool(user_groups & self._access_dashboard_roles)
+
+    def check_dashboard_access(self, user_id):
+        """
+        :return: None
+        :raise GroupworkAccessDeniedError: If user can't access dashboard
+        """
+        if not self.can_access_dashboard(user_id):
+            raise GroupworkAccessDeniedError("User can't access dashboard")
+
+    def is_user_ta(self, user_id, course_id):
+        """
+        :return: True if user is a TA for a gven course
+        """
+
+        granted_roles = {r["role"] for r in self.project_api.get_user_roles_for_course(user_id, course_id)}
+        allowed_roles = set(self.ta_roles)
+        return bool(allowed_roles & granted_roles)
+
+    def check_outsider_allowed(self, user_id, course_id):
+        """
+        :return: None
+        :raise GroupworkAccessDeniedError: If user can't access a groupwork item
+        if he is not part of the team.
+        """
+        if not self.is_user_ta(user_id, course_id):
+            raise GroupworkAccessDeniedError("User can't access this group work")
+
+    def get_organisation_filter_for_user(self, user_id, additional_filter=None):
+        """
+        :param Iterable[int] additional_filter:
+            Iterable of organization ids that are filtered additionally.
+        :return: Returns object that can be used if user_id has access to
+                 an organisation. This callable returns bool an expects to be
+                 called with organization id.
+        :rtype: Callable[[int], bool]
+        """
+        if self._can_user_access_all_orgs(user_id):
+            allowed_org_ids = None
+        else:
+            allowed_orgs = self.project_api.get_user_organizations(user_id)
+            allowed_org_ids = (org['id'] for org in allowed_orgs)
+        return self.OrganizationFilterObject(self, user_id, allowed_org_ids, additional_filter)
+
+    class OrganizationFilterObject(object):
+
+        """
+        A class that checks whether an user can access an organisation.
+
+        Contains two sets: one with organisations user has access to, and second
+        with organisations that user has explicitily filtered for.
+
+        Value None means "all organisations" --- since there can be a large
+        number of organisations, we avoid keeping "all organisations" that way.
+        """
+
+        def __init__(self, parent, user_id, allowed_org_ids, filter_org_ids=None):
+            self.parent = parent
```

Comment:
@jbzdak looks like it only needs `project_api` from the parent; maybe it would be better to pass project_api, or even have it injected by `ProjectAPIXblockMixin` (which should than be renamed to not contain `XBlock`).